watertap/examples/SOP/SOP_prop_pack.py

```diff
@@ -0,0 +1,543 @@
+#
+###############################################################################
+
+"""
+Simple property package for selective oil permeation
+"""
+
+from pyomo.environ import (
+    Constraint,
+    Var,
+    Param,
+    NonNegativeReals,
+    Suffix,
+)
+from pyomo.environ import units as pyunits
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    PhysicalParameterBlock,
+    StateBlockData,
+    StateBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+)
+from idaes.core.base.components import Component
+from idaes.core.base.phases import LiquidPhase
+from idaes.core.util.initialization import (
+    fix_state_vars,
+    revert_state_vars,
+    solve_indexed_blocks,
+)
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_unfixed_variables,
+)
+from idaes.core.util.exceptions import PropertyPackageError
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+from idaes.core.solvers import get_solver
+from watertap.core.util.scaling import transform_property_constraints
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+# this first class sets up the property package and creates a central parameter block
+@declare_process_block_class("SopParameterBlock")
+class SopParameterData(PhysicalParameterBlock):
+    CONFIG = PhysicalParameterBlock.CONFIG()
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super(SopParameterData, self).build()
+
+        self._state_block_class = SopStateBlock
+
+        # phases
+        self.Liq = LiquidPhase()
+
+        # components
+        self.H2O = Component()
+        self.oil = Component()
+
+        # ---default scaling---
+        self.set_default_scaling("temperature", 1e-2)
+        self.set_default_scaling("pressure", 1e-6)
+        self.set_default_scaling("visc_d_phase_comp", 1e3)
+        self.set_default_scaling("dens_mass_phase_comp", 1e-3)
+        self.set_default_scaling("dens_mass_phase", 1e-3)
+
+    @classmethod
+    def define_metadata(cls, obj):
+        """Define properties supported and units."""
+        obj.add_properties(
+            {
+                "flow_mass_phase_comp": {"method": None},
+                "temperature": {"method": None},
+                "pressure": {"method": None},
+                "visc_d_phase_comp": {"method": "_visc_d_phase_comp"},
+                "dens_mass_phase_comp": {"method": "_dens_mass_phase_comp"},
+                "mass_frac_phase_comp": {"method": "_mass_frac_phase_comp"},
+                "flow_vol_phase_comp": {"method": "_flow_vol_phase_comp"},
+                "flow_vol_phase": {"method": "_flow_vol_phase"},
+                "flow_mass_phase": {"method": "_flow_mass_phase"},
+                "dens_mass_phase": {"method": "_dens_mass_phase"},
+                "vol_frac_phase_comp": {"method": "_vol_frac_phase_comp"},
```

Comment:
This looks like one of those properties that you'll need to throw into `obj.define_custom_properties`. See here for an example: https://github.com/watertap-org/watertap/pull/937/files

Note, this might not be the only property that you'll need to move out of this dict and move into `define_custom_properties`

Reply:
I think this comment is no longer relevant. I'm going to resolve, but please reopen if I am wrong.